### DISCUSSION TITLE
Access RBD from outside the cluster

### DISCUSSION THREAD

**Discussion by M-Gregoire** [rook/rook#14152] [Q&A] [answered]
I have a Ceph cluster in healthy condition running in a K8S cluster thanks to Rook.
I'm trying to access one of the block storage from outside the cluster.

I've created a`ceph.conf` file containing:
```
[global]
mon_host = <node-ip>:6789

[client.admin]
keyring = ./keyring
```

and `keyring`:
```
[client.admin]
key = <REDACTED>
```

However, the port `6789` is not exposed from the server, leading to commands like
```
$ rbd -c ceph.conf ls blockpool
2024-05-02T12:27:15.581+0200 7ff97f231440  0 monclient(hunting): authenticate timed out after 300
```

to fail.

I wasn't able to find any documentation about the best way to expose the Ceph Cluster, and specifically the block storage, outside of the cluster.
Is there a recommended way to do so?

Thank you

**Comment by M-Gregoire** [comment, marked as answer]
I've just found: https://github.com/rook/rook/issues/561#issuecomment-330319885 and https://github.com/rook/rook/issues/3220

Enabling https://github.com/rook/rook/blob/8416befe1bd9346b841322a735396f75f3d8ca19/deploy/charts/rook-ceph/values.yaml#L582C1-L582C24 was the answer!

**Reply by travisn** [reply]
You'll want to use the `network: host` setting in the CephCluster CR, see also https://rook.io/docs/rook/latest/CRDs/Cluster/network-providers/#host-networking.

**Reply by travisn** [reply]
The host networking will cause the mon endpoints and OSDs to be on the host network, thus exposed to the external cluster.
